fetch.bs

```diff
@@ -278,66 +271,59 @@ for consistency.
 <p class="note no-backref">A <a for=/>header list</a> is essentially a
 specialized multimap. An ordered list of key-value pairs with potentially duplicate keys.
 
+<p>A <a for=/>header list</a> (<var>list</var>)
+<dfn export for="header list" lt="contains|does not contain">contains</dfn> a <a for=header>name</a>
+(<var>name</var>) if <var>list</var> contains a <a>header</a> whose <a for=header>name</a> is a
+<a>byte-case-insensitive</a> match for <var>name</var>.
+
 <p>To <dfn export for="header list" id=concept-header-list-append>append</dfn> a
-<a for=header>name</a>/<a for=header>value</a>
-(<var>name</var>/<var>value</var>) pair to a
-<a for=/>header list</a> (<var>list</var>), append a new
-<a>header</a> whose <a for=header>name</a>
-is <var>name</var>, <a lt="byte-lowercase">byte-lowercased</a>, and
-<a for=header>value</a> is <var>value</var>, to
-<var>list</var>.
+<a for=header>name</a>/<a for=header>value</a> (<var>name</var>/<var>value</var>) pair to a
+<a for=/>header list</a> (<var>list</var>), run these steps:
+
+<ol>
+ <li>
+  <p>If <var>list</var> <a for="header list">contains</a> <var>name</var>, then set <var>name</var>
+  to the first such <a>header</a>'s <a for=header>name</a>.
+
+  <p class="note no-backref">This reuses the casing of the <a for=header>name</a> of the
+  <a>header</a> already in the <a for=/>header list</a>, if any. If there are multiple matched
+  <a>headers</a> their <a for=header>names</a> will all be identical.
+  <!-- XXX Firefox and Safari adjust known header names too. -->
```

Comment:
I like the simplicity that provided case-insensitive header names to the spec.
Maybe a serialization context could have stored the casing information to leave the header names case-insensitive.
But anyway, this looks good to me as the PR is taking great care of making the specific casing unobservable from JS scripts.

I would tend to write the above statements as SHOULD/MUST/MAY, something like:
All matching headers must use the same casing.
The casing of the first header already in the list should be used for all matching headers.
The user agent may use its own casing for well-known headers.